src/app/dataset-view/additional-components/lineage-component/lineage.component.ts

```diff
@@ -108,11 +107,36 @@ export class LineageComponent extends BaseComponent implements OnInit {
                     isCurrent: dataset.id === currentDataset.id,
                     access: "private",
                     accountName: dataset.owner.name,
+                    sourceUrl:
+                        dataset.metadata.currentSource?.fetch.__typename === "FetchStepUrl"
+                            ? dataset.metadata.currentSource.fetch.url
+                            : "Other source",
                 },
             });
         }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        const extraNodes = this.lineageGraphNodes.filter((item: Node) => item.data.kind === DatasetKind.Root);
```

Comment:
Why can't you do this in the main loop above? Perhaps it's easier to work with the original data then later extracting them from nodes.

Reply:
Fixed